### DISCUSSION TITLE
Compiling pipper with pcaudiolib

### DISCUSSION THREAD

**Discussion by StoryHack** [rhasspy/piper#196] [Q&A] [answered]
I'm attempting to generate several .wavs using the json lines format, with "output_file" set in each json line. When I attempt to run piper using my .jsonl file, I get an error that says:
```
WARNING: Piper was not compiled with pcaudiolib. Output audio will be written to the current directory.
```
And all the wav files are indeed created with automatically generated names in the current directory. But I would really like to put the wavs with the names that get put in the directories I want. So, I'm tring to figure out how to compile piper with pcaudiolib.

I did put the  piper-phonemize stuff in the correct lib/ location. And I have installed a bunch of compiler-related packages according to the errors that running "make" was giving me. Now, when I run make, get the following output 

```
> make
mkdir -p build
cd build && cmake ../src/cpp -DCMAKE_BUILD_TYPE=Release && make
-- Configuring done
-- Generating done
-- Build files have been written to: /home/bryce/piper/build
make[1]: Entering directory '/home/bryce/piper/build'
make[2]: Entering directory '/home/bryce/piper/build'
make[3]: Entering directory '/home/bryce/piper/build'
Consolidate compiler generated dependencies of target piper
make[3]: Leaving directory '/home/bryce/piper/build'
make[3]: Entering directory '/home/bryce/piper/build'
[ 33%] Linking CXX executable piper
/usr/bin/ld: cannot find -lespeak-ng: No such file or directory
/usr/bin/ld: cannot find -lonnxruntime: No such file or directory
collect2: error: ld returned 1 exit status
make[3]: *** [CMakeFiles/piper.dir/build.make:113: piper] Error 1
make[3]: Leaving directory '/home/bryce/piper/build'
make[2]: *** [CMakeFiles/Makefile2:83: CMakeFiles/piper.dir/all] Error 2
make[2]: Leaving directory '/home/bryce/piper/build'
make[1]: *** [Makefile:91: all] Error 2
make[1]: Leaving directory '/home/bryce/piper/build'
make: *** [Makefile:9: piper] Error 2
```

I did install espeak and the onnxruntime, and I'm still gettig the same error. And I still don't know how to add in pcaudiolib.

What am I doing wrong?

**Comment by synesthesiam** [comment]
Did you run piper with the --json-input flag?

pcaudiolib is only used to play audio directly to the speakers. I don't compile it in by default because it pulls in a lot of other dependencies.

**Reply by StoryHack** [reply]
Here is the command:

```
cat ./tmp/cues.jsonl | ./piper/piper --model voice/libritts.onnx --json-input
```

And my jsonl file looks like:

```
{ "text": "Chapter 1", "speaker_id": 0, "output_file": "./tmp/1.wav" }
{ "text": "This is a test of json input.", "speaker_id": 1, "output_file": "./tmp/2.wav" }
{ "text": "I really hope I can get it work work.", "speaker_id": 1, "output_file": "./tmp/3.wav" }
```

The wavs are generated as if they are just lines of plain text, as I can hear it say "text" then the line, then some gibberishy stuff ending in "wav"

I get similar results when I try to echo a single line of json.  

```
echo '{ "text": "Welcome again to the world of speech synthesis!"}' | ./piper/piper --model voice/libritts.onnx speaker-id 0 --output_file welcomeagain.wav --json-input
```

I must be doing something wrong, but for the life of me, I can't figure out what.

**Comment by StoryHack** [comment, marked as answer]
Nevermind. Once again, I'm an idiot. Somehow I hadn't updated to the latest binary release. Once I did, everything worked as I wanted, and now I have no need of compiling. Thanks for putting up with me.

**Reply by synesthesiam** [reply]
No worries, glad it's working!
